specs/mgt-search-suggestion.md

```diff
@@ -0,0 +1,161 @@
+| search-suggestion-text | suggestionText[]: The text suggestion details list | The template to render text entity. |
+| search-suggestion-file | suggestionFile[]: The file suggestion details list | The template to render file entity. |
+
+The following examples shows how to use the `error` template.
+
+```html
+<mgt-search-suggestion>
+    <template data-type="search-suggestion-texts">
+        <div  data-for="text in texts">
+        {{text.text}}
+        </div>
+    </template>
+</mgt-search-suggestion>
+
+```
+
+## Proposed Solution
+
+### Example 1: basic usage without any callback function
+```<mgt-search-suggestion></mgt-search-suggestion>```
+
+### Example 2: Developer provides a site-id and item-id
+```<mgt-search-suggestion id="search-suggestion"> </mgt-search-suggestion>```
+```
+#User can get this component then bind callback action to this component 
+
+function onClickCallback(suggestionValue) {
+    console.log('suggestion value:', suggestionValue);
+    var searchValue = getSuggestionValue(suggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function onEnterKeyPressCallback(originalValue, selectedSuggestionValue) {
+    console.log('original value:', originalValue);
+    console.log('suggestion value:', selectedSuggestionValue);
+    var searchValue = getSuggestionValue(selectedSuggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function getSuggestionValue(suggestionValue) {
+    var searchValue = '';
+    if (suggestionValue.entity == 'File') {
+        searchValue = suggestionValue.name;
+    } else if (suggestionValue.entity == 'Text') {
+        searchValue = suggestionValue.text;
+    } else if (suggestionValue.entity == 'People') {
+        searchValue = suggestionValue.displayName;
+    }
+    return searchValue;
+}
+
+var obj = document.getElementById('search-suggestion');
+obj.onClickCallback = onClickCallback;
+obj.onEnterKeyPressCallback = onEnterKeyPressCallback;
+
+```
+
+### Example 3: Developer can select entity types
+
+```<mgt-search-suggestion selected-entity-types="file, text, people"></mgt-search-suggestion>```
+
+```<mgt-search-suggestion selected-entity-types="file, people"></mgt-search-suggestion>```
+
+```<mgt-search-suggestion selected-entity-types="people, text"></mgt-search-suggestion>```
+
+## Example 4: Developer can set each entity type's suggestion count
+
+```<mgt-search-suggestion max-text-suggestion-count="3" max-file-suggestion-count="2" max-people-suggestion-count="3"></mgt-search-suggestion>```
+
+## Attributes and Properties
+
+| Attribute | Property | Description |
+| --------- | -------- | ----------- |
+| `max-text-suggestion-count` | `maxTextSuggestionCount` | The max suggestion count for text. |
+| `max-people-suggestion-count` | `maxPeopleSuggestionCount` | The max suggestion count for people. |
+| `max-file-suggestion-count` | `maxFileSuggestionCount` | The max suggestion count for file. |
```

Comment:
Consider simplifying the property names:

```suggestion
| `max-query-suggestions` | `maxQuerySuggestions` | The max suggestion count for text. |
| `max-people-suggestions` | `maxPeopleSuggestions` | The max suggestion count for people. |
| `max-file-suggestions` | `maxFileSuggestions` | The max suggestion count for file. |
```

Reply:
I deleted them all, and the limitation merged into entity-types property